back/db/views/analytics_dimension_user_custom_field_values_v01.sql

```diff
@@ -0,0 +1,19 @@
+SELECT
+  DISTINCT u.id as dimension_user_id,
```

Comment:
Might be worth using a `GROUP_BY` instead of a `DISTINCT` if it's possible here. I think it's generally a lot more efficient

Reply:
I just tested it with this code on the tenant with the highest number of users (67394)
```ruby
Benchmark.measure do
  10.times do
    key_values = ActiveRecord::Base.connection.execute("SELECT
  u.id as dimension_user_id,
  cf.key,
  cf.value
FROM
  users u
  LEFT JOIN LATERAL (
    SELECT
      key,
      custom_field_values ->> key AS value
    FROM
      custom_fields
    where
      custom_fields.resource_type = 'User'
  ) cf ON true
  LEFT JOIN LATERAL (
    SELECT
      jsonb_object_keys(u.custom_field_values) AS key
  ) cfv ON true
  where cf.value = '0360'
GROUP BY
  dimension_user_id,
  cf.key,
  cf.value;").to_a; nil
  end
end

Benchmark.measure do
  10.times do
    ActiveRecord::Base.connection.execute("SELECT
  DISTINCT u.id as dimension_user_id,
  cf.key,
  cf.value
FROM
  users u
  LEFT JOIN LATERAL (
    SELECT
      key,
      custom_field_values ->> key AS value
    FROM
      custom_fields
    where
      custom_fields.resource_type = 'User'
  ) cf ON true
  LEFT JOIN LATERAL (
    SELECT
      jsonb_object_keys(u.custom_field_values) AS key
  ) cfv ON true
  where cf.value = '0360';").to_a; nil
  end
end
```
and it works the same `@real=1.18`. But I still changed it just to [cover more cases](https://stackoverflow.com/questions/70433926/postgresql-which-way-is-faster-distinct-or-group-by).